src/neo4j_genai/generation/rag.py

```diff
@@ -0,0 +1,61 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import logging
+from typing import Optional
+from ..retrievers.base import Retriever
+from .llm import LLMInterface
+from .prompts import RagTemplate
+
+
+logger = logging.getLogger(__name__)
+
+
+class RAG:
+    def __init__(
+        self,
+        retriever: Retriever,
```

Comment:
Do we need to create a type for input validation here? In which case, it can't be in the `types.py` file due to circular imports.

Reply:
Maybe we can have a `types.py` in each subfolder? I mean one for retrievers and one for generation?

Reply:
Yes that makes sense to me from a separations of concern POV